src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -284,11 +300,20 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionContext context)
                 Map<Digest, Message> additionalInputs = Maps.newHashMapWithExpectedSize(2);
                 additionalInputs.put(actionKey.getDigest(), action);
                 additionalInputs.put(commandHash, command);
+                Duration networkTimeStart = networkTime.getDuration();
+                Stopwatch uploadTime = Stopwatch.createStarted();
                 remoteCache.ensureInputsPresent(merkleTree, additionalInputs);
+                spawnMetrics.setUploadTime(
+                    uploadTime.elapsed().minus(
+                        networkTime.getDuration().minus(networkTimeStart)));
```

Comment:
Can you please explain why you are subtracting here?

Reply:
Because the tacit understanding in SpawnMetrics, evident in this [otherTime()](https://github.com/bazelbuild/bazel/blob/c35746d7f3708acb0d39f3082341de0ff09bd95f/src/main/java/com/google/devtools/build/lib/actions/SpawnMetrics.java#L162-L171) calculation, is an expectation of cumulative totalTime representing all component times, which are all expected to match the wall clock time calculations.

Time spent in one metric, i.e. the wall time spent uploading all inputs, must not include the time spent during another metric, in this case the time spent waiting for network responses, so that they can be used to determine a percentage of total wall clock time each metric represents.

Reply:
Added a comment detailing the reasoning here.